src/atomic.rs

```diff
@@ -496,7 +496,8 @@ impl<T> Atomic<T> {
     /// Bitwise "and" with the current tag.
     ///
     /// Performs a bitwise "and" operation on the current tag and the argument `val`, and sets the
-    /// new tag to the result. Returns the previous pointer.
+    /// new tag to the result. Returns the previous pointer.  Before the operation, `val` is
+    /// truncated to be fit in the bitmask returned by [`tag_bits()`](fn.tag_bits.html)).
```

Comment:
I'd rather not mention the `tag_bits()` method since it is not public.

My suggestion is to leave this comment as-is, and simply add the following before `# Examples` in the documentation for `Owned::with_tag` and `Ptr::with_tag`:

```rust
/// # Panics
///
/// Panics if the tag doesn't fit into the unused bits of an aligned pointer to `T`.
```

@Vtec234  said:

> Please mention in the documentation of all three functions that the tags are truncated to a valid size.

But the documentation here already states: *Performs a bitwise "and" operation on the current tag and the argument `val`*, so I think truncation is already implied.